pyccel/ast/numpyext.py

```diff
@@ -579,22 +630,39 @@ def stop(self):
         return self._stop
 
     @property
-    def size(self):
-        return self._size
+    def num(self):
+        """Represent the number of generated elements by the linspace function."""
+        return self._num
 
     @property
     def index(self):
         return self._index
 
+    @property
+    def rank(self):
+        return self._rank
+
     @property
     def step(self):
-        return (self.stop - self.start) / (self.size - 1)
+        if isinstance(self.endpoint, LiteralFalse):
+            return PyccelDiv(PyccelMinus(self.stop, self.start), self.num)
+        elif isinstance(self.endpoint, LiteralTrue):
+            PyccelDiv(PyccelMinus(self.stop, self.start), PyccelMinus(self.num, LiteralInteger(1)))
+        return PyccelDiv(PyccelMinus(self.stop, self.start), PyccelMinus(self.num, PythonInt(self.endpoint)))
 
-    def __str__(self):
-        code = 'linspace({}, {}, {})'.format(str(self.start),
-                                             str(self.stop),
-                                             str(self.size))
-        return code
+    @property
+    def ind(self):
+        """Used to store the indice needed for the linspace function."""
+        return self._ind
```

Comment:
Is `ind` equivalent to `self._print(index)`?

Reply:
yes it is

Reply:
Why do you need to store that?

Reply:
because i need the name of the variable generated by the loops to use it in the ifternaryoperator

Reply:
Can you point me to the code? I can't think of anytime where this information would need to be known outside of the printing stage and at the printing stage it can be retrieved with `self._print` as is done for everything else, but maybe I am missing something?